app/routes.py

```diff
@@ -1,2 +1,197 @@
+    
+    db.session.add(new_task)
+    db.session.commit()
+
+    return jsonify({"task": new_task.to_json()}), 201
+
+
+@tasks_bp.route("", methods=["GET"])
+def get_task():
+    tasks = Task.query.all()
+    
+    sort = request.args.get("sort")
+    if sort == "asc":
+        tasks = Task.query.order_by(Task.title).all()
+    elif sort == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+
+    tasks_response = [task.to_json() for task in tasks]
+    # for task in tasks:
+    #     tasks_response.append(task.to_json())
+
+    return jsonify(tasks_response), 200
+
+
+@tasks_bp.route("/<int:task_id>", methods=["GET", "PUT", "DELETE"])
+def get_single_task(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response("", 404)
+
+    if request.method == "GET":
+        return jsonify({"task": task.to_json()}), 200
+
+    elif request.method == "PUT":
+        request_body = request.get_json()
+        task.title = request_body["title"]
+        task.description = request_body["description"]
+
+        db.session.commit()
+
+        return jsonify({"task": task.to_json()}), 200
+
+    elif request.method == "DELETE": 
+        db.session.delete(task)
+        db.session.commit()
+
+        return jsonify({
+            "details": f'Task {task_id} \"{task.title}" successfully deleted'
+        })
+
+
+@tasks_bp.route("/<int:task_id>/<complete>", methods=["PATCH"])
+def update_completed_at(task_id, complete):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
+
+    if complete == "mark_complete":
+        task.completed_at = datetime.utcnow()
+
+        # db.session.commit()
+
+        slack_url = "https://slack.com/api/chat.postMessage"
+        headers = {
+            'Content-Type': 'application/json',
+            "Authorization": f"Bearer {slack_token}"
+        }
+        slack_data = {
+            "channel": "C021RGYNY48",
+            "text": f"Someone just completed the task {task.title}"
+        }
+
+        requests.post(slack_url, json=slack_data, headers=headers)
+
+    elif complete == "mark_incomplete":
+        task.completed_at = None
+
+    db.session.commit()
+
+    return jsonify({"task" :task.to_json()}), 200
+
+
+@goals_bp.route("", methods=["POST"])
+def add_goal():
+    request_body = request.get_json()
+    title = request_body.get("title")
+
+    if not title:
+        return jsonify({"details": "Invalid data"}), 400
+
+    new_goal = Goal(title=title)
+    
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return jsonify({"goal": new_goal.to_json()}), 201
+
+
+@goals_bp.route("", methods=["GET"])
+def get_goal():
+    goals = Goal.query.all()
+    goals_response = [goal.to_json() for goal in goals]
+
+    # for goal in goals:
+    #     goals_response.append(goal.to_json())
+
+    return jsonify(goals_response), 200
+
+
+@goals_bp.route("/<int:goal_id>", methods=["GET", "PUT", "DELETE"])
```

Comment:
In general I suggest you be consistent and have either all the functions handle one path/method each (preferred), or have all the functions handle a specific route with if-else blocks for each of the methods.